packages/contracts/contracts/DependencyRegistryV0.sol

```diff
@@ -742,7 +821,20 @@ contract DependencyRegistryV0 is
             return "";
         }
 
-        return _readFromBytecode(dependency.scriptBytecodeAddresses[_index]);
+        address scriptAddress = dependency.scriptBytecodeAddresses[_index];
+        bytes32 storageVersion = BytecodeStorageReader
+            .getLibraryVersionForBytecode(scriptAddress);
+
+        if (storageVersion == BytecodeStorageReader.UNKNOWN_VERSION_STRING) {
+            return
```

Comment:
wondering if it would make sense to add this fallback logic into the bytecodestorage library itself, or just keep here?

Reply:
I don't love introducing this logic into the library itself, given that there isn't a great way to do on-chain introspection into a contract and _know_ that it was written with `SSTORE2`, so I like the design decision of pushing the specific fallback handling for `SSTORE2` reads onto the library client contract (such as this one), but could be definitely convinced otherwise (if you feel strongly, or if we got strong feedback from other prospective library clients in the future).

Reply:
thanks for the solid points, and I agree with what you are saying here 🙏 SGTM!